.github/workflows/main.yml

```diff
@@ -90,14 +111,377 @@ jobs:
 
       - run: yarn webpack --test --no-lavamoat --no-cache --browser=chrome --lockdown --sentry --snow --env production
         env:
-          SEGMENT_PROD_WRITE_KEY: '-' # A fake key just to make the build succeed
+          SEGMENT_PROD_WRITE_KEY: ${{ secrets.SEGMENT_PROD_WRITE_KEY }}
 
       - name: Upload artifact prep-build-test-webpack
         uses: actions/upload-artifact@v4
         with:
           name: prep-build-test-webpack
           path: ./dist/
 
+  prep-build-test:
+    needs: prep-deps
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout and setup environment
+        uses: metamask/github-tools/.github/actions/checkout-and-setup@1299bb1de0c6974ae6d0a32c7e8897fe168239ac
+        with:
+          is-high-risk-environment: false
+
+      - run: yarn build:test
+
+      - name: Move test build to dist-test
+        run: mv ./dist ./dist-test
+
+      - name: Move test zips to builds-test
+        run: mv ./builds ./builds-test
```

Comment:
The majority of this `mv` stuff in the `prep-build` steps is probably unnecessary.  It was acrobatics to deal with the way CircleCI workspaces function.  In GitHub Actions, it's probably not needed.